File: apps/auth-server/src/main/java/com/example/chat/auth/server/core/domain/policy/AuthPolicy.java
package com.example.chat.auth.server.core.domain.policy;

import com.example.chat.auth.server.core.domain.AuthLevel;
import com.example.chat.auth.server.core.domain.AuthenticationContext;
import com.example.chat.auth.server.core.domain.MfaRequirement;

/**
 * 인증 정책
 */
public interface AuthPolicy {
    MfaRequirement checkMfaRequirement(AuthenticationContext context, String sessionId);

    AuthLevel requiredAuthLevel(String operation);
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/core/domain/policy/AuthPolicy.kt
package com.example.chat.auth.server.core.domain.policy

import com.example.chat.auth.server.core.domain.AuthLevel
import com.example.chat.auth.server.core.domain.AuthenticationContext
import com.example.chat.auth.server.core.domain.MfaRequirement

/** 인증 정책 */
interface AuthPolicy {
    fun checkMfaRequirement(context: AuthenticationContext, sessionId: String): MfaRequirement
    fun requiredAuthLevel(operation: String): AuthLevel
}
